src/hugr/validate.rs

```diff
@@ -393,7 +393,7 @@ impl<'a> ValidationContext<'a> {
                 true
             }
             ty => {
-                if !local && !matches!(ty, EdgeKind::Value(SimpleType::Classic(_))) {
+                if !local && !matches!(&ty, EdgeKind::Value(t) if t.least_upper_bound().is_some()) {
```

Comment:
At this point it may be clearer to separate the `EdgeKind::Value` match branch (even if it needs repeating the return statement).
And maybe add a comment that `least_upper_Bound = None ⟹ non-classical`.